test/spmd/tensor/parallel/test_parallelize_api.py

```diff
@@ -71,6 +79,55 @@ def test_creat_1d_device_mesh_error(self):
         ):
             _create_1d_device_mesh(mesh, 3)
 
+    def _check_module(
+        self,
+        local_module,
+        dist_module,
+        local_input,
+        dist_input,
+        device_mesh,
+        local_optim,
+        dist_optim,
+    ):
+        replicate = [Replicate()]
+
+        # ensure the parameter is properly distributed.
+        # TODO: see how to make it work on `mlp`: nested modules
+        for name, param in local_module.named_parameters():
+            dist_param = dist_module.get_parameter(name)
+            self.assertEqual(
+                param,
+                dist_param.redistribute(
+                    device_mesh=device_mesh, placements=replicate
+                ).to_local(),
+            )
+
+        # clear gradients
+        local_optim.zero_grad()
+        dist_optim.zero_grad()
+
+        # check forward correctness
+        local_output = local_module(local_input)
+        dist_output = dist_module(dist_input)
+        self.assertEqual(local_output, dist_output.to_local())
+
+        local_output.sum().backward()
+        dist_output.sum().backward()
+
+        # check backward correctness:
+        #   ensure gradients are same
+        for name, param in local_module.named_parameters():
+            dist_param = dist_module.get_parameter(name)
+            self.assertEqual(
+                param.grad,
+                dist_param.grad.redistribute(
+                    device_mesh=device_mesh, placements=replicate
+                ).to_local(),
+            )
```

Comment:
I believe this can be further consolidated since it looks very similar with L96-L103. Let's probably extract this logic and make it a util func, so that you can make it work for mlp? (For mlp, you can just have a for loop and call this func inside the for loop?

Reply:
that makes sense. is `test/spmd/tensor/parallel/utils.py` a good place to put the func?

Reply:
added TODO.